pandas/core/indexes/base.py

```diff
@@ -1376,6 +1393,12 @@ def set_names(self, names, level=None, inplace: bool = False):
         if not is_list_like(names) and level is None and self.nlevels > 1:
             raise TypeError("Must pass list-like as `names`.")
 
+        if is_dict_like(names) and not isinstance(self, ABCMultiIndex):
+            raise TypeError("Can only pass dict-like as `names` for MultiIndex.")
+
+        if is_dict_like(names) and level is not None:
+            raise TypeError("Can not pass level when passing dict-like as `names`.")
+
```

Comment:
doesn't this hit a dict-like names?

Reply:
Yikes, yes you are right. have not though about dict like names. Removed this